sim/scenarios/droplist/droplist-error-model.cc

```diff
@@ -18,9 +19,12 @@ DroplistErrorModel::DroplistErrorModel()
 void DroplistErrorModel::DoReset(void) { }
  
 bool DroplistErrorModel::DoCorrupt(Ptr<Packet> p) {
-    if(drops.find(++packet_num) == drops.end())
-        return false;
-    cout << "Dropping packet number " << packet_num << endl;
+    if(!IsUDPPacket(p)) return false;
+    if(drops.find(++packet_num) == drops.end()) return false;
+    
+    QuicPacket qp = QuicPacket(p);
+    cout << "Dropping packet " << packet_num << " (" << qp.GetUdpPayload().size() << " bytes) from " << qp.GetIpv4Header().GetSource() << endl;
```

Comment:
This introduces a requirement for the packet to be an IPv4 packet, right? Why is this needed?

Reply:
That's a general requirement of `QuicPacket` at the moment (which is why I added the TODO). We'll need a general fix for that (any suggestions for the class API?). Just added it here to get some more useful logging here.

Reply:
Understood. I would suggest that QuicPacket can be a class that handles both v4 and v6 packets; a "QUIC packet" ought to be independent of IP version.

One way to do this might be to have the QuicPacket constructor do the same check you do now in IsUDPPacket() and store v4 or v6 address. You can then have QuicPacket.GetIpv4Address() and QuicPacket.GetIpv6Address() methods that either return empty string or the address, and a QuicPacket.GetIpAddress() method that returns whichever one is populated.

This doesn't have to be done in this PR, but given that we're trying to support v6, it might be a good idea to not introduce more v4-only dependencies.